modules/frontend/src/components/pages/DataQualityPage.vue

```diff
@@ -5,14 +5,47 @@
       <div class="container-fluid">
         <div class="ms-auto d-flex align-items-center text-white">
           <i class="bi bi-person-fill fs-4 me-2"></i>
-          <span>{{ username }}</span>
+          <!-- Warning indicator for default password -->
+          <div v-if="defaultPasswordFlag" class="position-relative me-2"
+               @mouseenter="showTooltip = true"
+               @mouseleave="showTooltip = false">
+            <i class="bi bi-exclamation-triangle-fill text-warning fs-5"></i>
+            <div v-if="showTooltip" class="custom-tooltip">
+              You are using the default password. Please change it for security.
+            </div>
+          </div>
+
+          <div class="position-relative">
+            <span class="username-dropdown cursor-pointer" @click="toggleDropdown">
+              {{ username }}
+              <i class="bi bi-chevron-down ms-1"></i>
+            </span>
+
+            <div v-if="showDropdown" class="dropdown-menu-custom position-absolute">
+              <button class="dropdown-item-custom" @click="showChangePasswordModal">
+                <i class="bi bi-key me-2"></i>
+                Change Password
+                <span v-if="defaultPasswordFlag" class="badge bg-warning text-dark ms-2">Required</span>
+              </button>
+            </div>
+          </div>
           <button class="btn text-white ms-3" @click="logout">
             <i class="bi bi-box-arrow-right"></i> Logout
           </button>
         </div>
       </div>
     </nav>
 
+    <PasswordChangeModal :isVisible="showPasswordModal" @close="closePasswordModal" />
+
+    <!-- Default Password Alert -->
+    <div v-if="defaultPasswordFlag" class="alert alert-warning alert-dismissible fade show mx-4 mt-3" role="alert">
+      <i class="bi bi-exclamation-triangle-fill me-2"></i>
+      <strong>Security Warning:</strong> You are using the default password. Please change it immediately for security reasons.
```

Comment:
I would remove this alert as the warning next to the username looks sufficient to me.